### PR TITLE
Add unix socket support

### PR DESCRIPTION
**Pull request by stepankuzmin**
Hi there! I've add socket support, so tessera can listen on unix socket. It might be useful when running multiple tessera instances.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -100,9 +100,23 @@ module.exports = function(opts, callback) {
     });
   }
 
-  app.listen(process.env.PORT || opts.port, process.env.HOST || opts.bind, function() {
-    console.log("Listening at http://%s:%d/", this.address().address, this.address().port);
+  var handler = process.env.SOCKET || opts.socket || process.env.PORT || opts.port;
+  var server = app.listen(handler, process.env.HOST || opts.bind, function() {
+    var endpoint;
+    if (opts.socket) {
```
Comment (mojodna) [inline server.js]: Should this also apply to `process.env.SOCKET`?

File: server.js [hunk 2]
```diff
@@ -100,9 +100,23 @@ module.exports = function(opts, callback) {
...
+      endpoint = opts.socket;
+      fs.chmodSync(opts.socket, '1766');
```
Comment (mojodna) [inline server.js]: Is this a `umask` artifact, or is this standard when listening on sockets?

**Comment by mojodna** [comment]
Nice, I like it.

File: server.js (hunk 2 above)
Comment (stepankuzmin) [inline server.js]: Yep, it's `umask` to ensure that socket can be accessed from other users/groups, so you don't have to add nginx user, for example, to the group what runs tessera.

File: server.js (hunk 1 above)
Comment (mojodna) [inline server.js]: Bump.

Comment (stepankuzmin) [inline server.js]: I think it would be logical to indicate this in `process.env` too, but this is optional.

**Pull request merged**